src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -369,8 +369,17 @@ def column_states(self) -> ColumnStateDict:
         Refer to class ColumnStateDict.
         """
         if self._column_states is None:
+            if self.analyzer.session.reduce_describe_query_enabled:
+                # data types are not needed in SQL simplifier, so we
+                # just create dummy data types here.
+                column_attrs = [
+                    Attribute(q, DataType())
```

Comment:
I'd prefer to use this "dummy" approach instead of removing data types by converting `list[Attribute]` to `list[str]`, which will result in massive code changes (because other code relies on the fact that `column_states.project` is  `list[Attribute]`, e.g., https://github.com/snowflakedb/snowpark-python/blob/47cadd26f25781eeb7898f9e291fe83132b1cf4e/src/snowflake/snowpark/dataframe.py#L4208-L4213). They are not easily parameter protected.

Reply:
> so for column expression, especially the Star(), does it require the correct DataType, or any data type is fine?
I don't think a correct data type is needed, but we need a data type object here because of alias map https://github.com/snowflakedb/snowpark-python/blob/8439f86474b1117bd3d8a73d0151aabd5466c054/src/snowflake/snowpark/_internal/analyzer/analyzer.py#L368-L371. I agree that having such `_output` method is confusing and error prone. 

> by introducing a new _output methods to selectStatement, when the flag is true, we just read the quoted_identifier and construct list of attributes with DataType, otherwise return the original projection.
I think the above implementation is exactly doing what you describe here? So how about having this implementation for now, and let's create another ticket to see whether we can refactor and remove `_output` method later? cc @sfc-gh-yixie